### ISSUE TITLE
uBlockOrigin settings page should get a "copy to clipboard" button

### ISSUE THREAD

**Issue by moviuro** [uBlockOrigin/uBlock-issues#1701]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

When trying to exchange with other projects (think: AdGuard), I need to share my uBlockOrigin config. The "Filter Lists" tab in the dashboard ([on Firefox](moz-extension://1759ced2-0a80-49d9-84ec-4864029907fe/dashboard.html#3p-filters.html)) does not have an easy way of copying the enabled/disabled filters.

### A specific URL where the issue occurs

moz-extension://1759ced2-0a80-49d9-84ec-4864029907fe/dashboard.html#3p-filters.html

### Steps to Reproduce

Navigate to "Filter Lists" tab in uBO settings

### Expected behavior

1. New button "Copy current config"
2. Can copy the current config in plaintext or markdown

This would be a very simple extract of all info available on the page + some info about each list:

* Setting toggles On/Off (Auto-update,... My filters,... uBlock filters - Privacy, etc.)
* URL/homepage of the lists
* Last update date / Time updated
* Version

Example:

```
[x] Auto-update filter lists
[x] Parse and enforce cosmetic filters
[ ] Ignore generic cosmetic filters

[x] My filters (1 used out of 1)

# Built-in

[x] uBlock filters (https://github.com/uBlockOrigin/uAssets, Last imported: 2020-07-08) (29573 used out of 29625)
[x] uBlock filters - badware risk (https://github.com/uBlockOrigin/uAssets, Last imported: ?) (2119 used out of 2120)
...

# Annoyances

[x] AdGuard Annoyances (https://github.com/AdguardTeam/AdGuardFilters, TimeUpdated: 2021-08-28T12:30:39+00:00) (43474 used out of 44135)
...
```

### Actual behavior

No button. Resort to manual copy (slow) or screenshot (creative way of wasting bandwidth).

### uBlock Origin version

1.37.2

### Browser name and version

Firefox ESR 78.13.0esr, Firefox 91.0.2

### Operating System and version

Windows 10, Archlinux

**Comment by felix-22** [comment]
Yeah, I have been thinking about the same long time ago already, as I noticed that filling issues and reporting ads etc on most issue trackers requires to provide some additional info like additional filter lists etc, for example uAssets: 

> Settings: List all the changes you've made to uBO's default settings here

- it would be much easier if there have been a simple copy button to gather all the modifications which have been made compared to default settings, and then a reporter can simply and quickly paste it into an issue template
- I often have seen reporters pasting screenshoots with their additional filter lists..which requires unnecessary extra work and time effort
- the same copy button is already implemented in uBO logger

**Comment by peace2000** [comment]
> I often have seen reporters pasting screenshoots with their additional filter lists..what requires unnecessary extra work and time effort and drains bandwich

Yeah, when someone files an issue report sharing a screenshot of filters in use and there are a lot custom filter lists that aren't uBO's stock lists, it's very laborious to check all them through as you'll have to google them. So true.

uBO currently has a function to export all it's settings (filterlists included) to a file and import as well but that's maybe a bit too hard to accomplish for many non-advanced users?

**Comment by felix-22** [comment]
> peace2000 : uBO currently has a function to export all it's settings (filterlists included) to a file and import as well 

Yeah, I'm aware of the export function, because I use it regularly to make backups.

> peace2000 : but that's maybe a bit too hard to accomplish for many non-advanced users?

The backup file has its own formatting which makes it a bit difficult to read as a plain text file, it is more suitable for exporting / importing than for normal reading.

I rarely see the backup file attached to issue template, most often the submitter just manually writes their filter lists or attaches screenshots in the issue, just wondering if attaching a backup is more difficult than attaching a screenshoot? Let's see:

- steps to create a copy/paste (2 steps):
   - press copy in uBO
   - press paste in github issue
   - 
- steps to create a screenshoot (2-4 steps):
   - press a hot-key
   - scroll down and make another screenshoot, so:
   - press a hot-key
   - drag the screenshoots from desktop to github issue
   - 
- steps to create a backup (10 steps):
   - scroll down
   - press create backup
   - press ok
   - compress the backup to zip (the screenshoots are already compressed)
   - drag the backup from desktop to github issue
   - the backup must be downloaded by filter maintentainers
   - it must be unzipped
   - scroll down in uBO
   - press import in uBO
   - press Open

**Comment by gwarser** [comment]
> Yeah, when someone files an issue report sharing a screenshot of filters in use and there are a lot custom filter lists that aren't uBO's stock lists, it's very laborious to check all them through as you'll have to google them. So true.

Custom configs are not really supported on uAssets.

**Comment by felix-22** [comment]
> gwarser : Custom configs are not really supported on uAssets.

The issue template says otherwise, so I'm, puzzled, could you clarify?:
![44](https://user-images.githubusercontent.com/75935068/131667058-857fbe7b-013b-4e9d-a2d2-ae9dbf71086d.png)

Or by "custom configs" do you mean the import/export backup feature only or in general?
Since you quoted the screenshoots method, I'm not sure what exactly you mean.

**Comment by uBlock-user** [comment]
> The issue template says otherwise, 

That asks you to list the settings,

Once found that the issue is caused by non-default list we proceed to decline it.

**Comment by felix-22** [comment]
> uBlock-user : That asks you to list the settings

Nope, that asks to list **changes**, but I understand what you mean, so that's just a spelling mistake in the template.

**Comment by felix-22** [comment]
uBlock-user : Once found that the issue is caused by non-default list we proceed to decline it.

Oh, this is a game breaker then, I it seems the issue is going to be declined and the submitters to be told to report issues with default filter lists.

**Comment by uBlock-user** [comment]
> No, that asks to list "changes (- change 1, - change 2, - change 3)",

Placeholder, lets not argue semantics.


> it seems the feature request is going to be labeled as invalid and declined then

Upto to @gorhill

**Comment by krystian3w** [comment]
> AdGuard's filters lists but they are not enabled by default

Few countries based on regional AdGuard Lists e.g. Japan, Turkey, France/Arabic. But AG germany was rejected due fastly fix pages by `okiehsch`/ `mapx-`.

Other story is enforce first report Annoyances to:

- Easylist - started add CSS/Scriptlets with uBO syntax (mabye try learn use `denyallow=` /create safe `csp=`  blockades)
- AdGuard Annoyances - started support `:style(`  very long time ago and actively try create scriptlets compatible with uBO.

**Comment by TRSx80** [comment]
In the meantime, one workaround could be to run the full `my-ublock-backup_YYYY-MM-DD_HH.MM.DD.txt` file through something like `jq`, in order to get whatever part(s) you might want?  I think it could easily be implemented as a one-liner even.

**Comment by felix-22** [comment]
Put the log into code tag with syntax highlighting? Example: https://github.com/uBlockOrigin/uBlock-issues/issues/1701#issuecomment-943565854
Doc: https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting
